pkg/controller/recommendation/recommendation_trigger_controller.go

```diff
@@ -0,0 +1,160 @@
+	}
+
+	identities := map[string]ObjectIdentity{}
+	key := objRefKey(recommendation.Spec.TargetRef.Kind, recommendation.Spec.TargetRef.APIVersion, recommendation.Spec.TargetRef.Namespace, recommendation.Spec.TargetRef.Name)
+	identities[key] = ObjectIdentity{
+		Namespace:  object.GetNamespace(),
+		Name:       object.GetName(),
+		Kind:       object.GetKind(),
+		APIVersion: object.GetAPIVersion(),
+		Labels:     object.GetLabels(),
+		Object:     *object,
+	}
+
+	newStatus := recommendationRule.Status.DeepCopy()
+	currentMissionIndex := -1
+	for index, mission := range newStatus.Recommendations {
+		if mission.UID == recommendation.UID {
+			currentMissionIndex = index
+			break
+		}
+	}
+
+	if currentMissionIndex == -1 {
+		klog.Warningf("cannot found recommendation mission %s", recommendationRuleRef.Name)
+		return ctrl.Result{}, nil
+	}
+
+	executeMission(context.TODO(), nil, c.RecommenderMgr, c.Provider, c.PredictorMgr, recommendationRule, identities, &newStatus.Recommendations[currentMissionIndex], recommendation, c.Client, c.ScaleClient, metav1.Now(), newStatus.RunNumber)
+	if newStatus.Recommendations[currentMissionIndex].Message != "Success" {
+		err = c.Client.Delete(context.TODO(), recommendation)
+		if err != nil {
+			klog.Errorf("Failed to delete recommendation %s when checking: %v", klog.KObj(recommendation), err)
+		}
+	}
+
+	updateRecommendationRuleStatus(context.TODO(), c.Client, c.Recorder, recommendationRule, newStatus)
+
+	return ctrl.Result{}, nil
+}
+
+func (c *RecommendationTriggerController) SetupWithManager(mgr ctrl.Manager) error {
+	c.discoveryClient = discovery.NewDiscoveryClientForConfigOrDie(mgr.GetConfig())
+	c.dynamicClient = dynamic.NewForConfigOrDie(mgr.GetConfig())
+
+	controller, err := controller.New("recommendation-trigger-controller", mgr, controller.Options{
+		Reconciler: c})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to Recommendation that runNumber decrease
+	return controller.Watch(&source.Kind{Type: &analysisv1alpha1.Recommendation{}}, &recommendationEventHandler{
+		enqueueHandler: handler.EnqueueRequestForObject{},
+	})
+}
+
+type recommendationEventHandler struct {
+	enqueueHandler handler.EnqueueRequestForObject
+}
+
+func (h *recommendationEventHandler) Create(evt event.CreateEvent, q workqueue.RateLimitingInterface) {
+}
+
+func (h *recommendationEventHandler) Delete(evt event.DeleteEvent, q workqueue.RateLimitingInterface) {
+}
+
+func (h *recommendationEventHandler) Update(evt event.UpdateEvent, q workqueue.RateLimitingInterface) {
+	newRecommendation := evt.ObjectNew.(*analysisv1alpha1.Recommendation)
+	oldRecommendation := evt.ObjectOld.(*analysisv1alpha1.Recommendation)
+	klog.V(6).Infof("recommendation %s OnUpdate", klog.KObj(newRecommendation))
+
+	oldRunNumber, _ := utils.GetRunNumber(oldRecommendation)
+	newRunNumber, _ := utils.GetRunNumber(newRecommendation)
```

Comment:
the error is not handled

Reply:
if get run number failed , we treat it as a empty number which by default is 0.